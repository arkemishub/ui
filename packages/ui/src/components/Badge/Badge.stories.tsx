/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Badge } from "./index";
import React from "react";
import { IBadgeProps } from "./Badge.types";

export default {
  title: "Badge",
  component: Badge,
};

export const Default = (args: Partial<IBadgeProps>) => {
  return (
    <Badge label={""} {...args}>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth="1.5"
        stroke="currentColor"
        className="w-6 h-6"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          d="M21.75 6.75v10.5a2.25 2.25 0 01-2.25 2.25h-15a2.25 2.25 0 01-2.25-2.25V6.75m19.5 0A2.25 2.25 0 0019.5 4.5h-15a2.25 2.25 0 00-2.25 2.25m19.5 0v.243a2.25 2.25 0 01-1.07 1.916l-7.5 4.615a2.25 2.25 0 01-2.36 0L3.32 8.91a2.25 2.25 0 01-1.07-1.916V6.75"
        />
      </svg>
    </Badge>
  );
};
