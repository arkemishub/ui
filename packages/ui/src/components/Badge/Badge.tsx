/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use client";

import React from "react";
import { useState } from "react";
import { twMerge } from "tailwind-merge";
import type { IBadgeProps } from "./Badge.types";

function Badge({ className, color, label, children }: IBadgeProps) {
  return (
    <div className="relative">
      <div
        data-testid="arke-badge"
        className={twMerge(
          "badge",
          color === "primary" && "badge--primary",
          color === "secondary" && "badge--secondary",
          className
        )}
      >
        {label}
      </div>
      {children}
    </div>
  );
}

export default Badge;
