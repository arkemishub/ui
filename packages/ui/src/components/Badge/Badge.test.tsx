/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Badge component test

import Badge from "./Badge";
import { render, screen } from "@testing-library/react";

describe("Badge", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<Badge>Badge</Badge>);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Badge component", () => {
    render(<Badge>Badge</Badge>);
    expect(screen.getByTestId("arke-badge")).toBeInTheDocument();
  });

  test("should render label", () => {
    render(<Badge label="Label">Badge</Badge>);
    expect(screen.getByText("Label")).toBeInTheDocument();
  });

  test("should render primary badge", () => {
    render(<Badge color="primary">Badge</Badge>);
    expect(screen.getByTestId("arke-badge")).toHaveClass("badge--primary");
  });

  test("should render secondary badge", () => {
    render(<Badge color="secondary">Badge</Badge>);
    expect(screen.getByTestId("arke-badge")).toHaveClass("badge--secondary");
  });

  test("should add custom class", () => {
    render(<Badge className="custom-class">Badge</Badge>);
    expect(screen.getByTestId("arke-badge")).toHaveClass("custom-class");
  });
});
