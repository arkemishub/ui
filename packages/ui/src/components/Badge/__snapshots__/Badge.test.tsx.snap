// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="relative"
  >
    <div
      class="badge"
      data-testid="arke-badge"
    />
    Badge
  </span>
</DocumentFragment>
`;
